pkg/apis/types/submit_pytorchjob.go

```diff
@@ -45,4 +45,8 @@ type SubmitPyTorchJobArgs struct {
 
 	// ShareMemory Specifies the shared memory size
 	ShareMemory string `yaml:"shareMemory"`
+
+	// Number of workers per node, supported values: [auto, cpu, gpu, int].
+	// For more, https://github.com/pytorch/pytorch/blob/26f7f470df64d90e092081e39507e4ac751f55d6/torch/distributed/run.py#L629-L658.
+	NprocPerNode string `yaml:"nprocPerNode,omitempty"`
```

Comment:
I suggest using  numbers for NprocPerNode is a sensible choice given its literal meaning, as it explicitly specifies the number of processes per node.  And by default it's 1 to make consistency with previous version.

Reply:
I think we should keep the `--nproc-per-node` the same way as `torchrun`:

```bash
$ torchrun --help
...
options:
  -h, --help            show this help message and exit
  --nnodes NNODES       Number of nodes, or the range of nodes in form <minimum_nodes>:<maximum_nodes>.
  --nproc-per-node NPROC_PER_NODE, --nproc_per_node NPROC_PER_NODE
                        Number of workers per node; supported values: [auto, cpu, gpu, int].
...
```

The available values are `auto`, `cpu`, `gpu` and an integer. If `--nproc-per-node` is unset, then `1` will be used, ref: https://github.com/pytorch/pytorch/blob/a8d6afb511a69687bbb2b7e88a3cf67917e1697e/torch/distributed/run.py#L433-L440.

For `arena`, if `--nproc-per-node` is unset, then we will not ingest env `PET_NPROC_PER_NODE` to the pod template. Otherwise, we will check whether the value specified by `--nproc-per-node` is valid and append it to the pod template envs.